src/components/input-dropdown/input-dropdown.component.spec.ts

```diff
@@ -237,6 +239,38 @@ describe('InputDropdownComponent', () => {
         expect(openChangeSpy).toHaveBeenCalledWith(false);
     });
 
+    it('should not open the dropdown when disabled is true', async () => {
+        let trigger = nativeElement.querySelector('button.form-control') as HTMLButtonElement;
+        component.disabled = true;
+
+        // open menu
+        trigger.click();
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        const dropdown = nativeElement.querySelector('.filter-container');
+        expect(dropdown).toBeNull();
+    });
+
+    it('should not retain focus on the button when disabled is true.', async () => {
+        let trigger = nativeElement.querySelector('button.form-control') as HTMLButtonElement;
+        component.disabled = true;
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        // open menu
+        trigger.click();
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        const button = document.querySelector<HTMLInputElement>('button.form-control');
+
+        expect(document.activeElement).not.toBe(button);
+    });
+
+
```

Comment:
Could you add a test checking the value of dropdownOpen before and after clicking the icon (the problem I mentioned in the previous review).